pkg/daemon/update.go

```diff
@@ -702,6 +691,61 @@ func (dn *Daemon) updateOS(config *mcfgv1.MachineConfig) error {
 	return nil
 }
 
+// getPendingConfig loads the JSON state we cache across attempting to apply
+// a config+reboot.  If no pending state is available, ("", nil) will be returned.
+// The bootID is stored in the pending state; if it is unchanged, we assume
+// that we failed to reboot; that for now should be a fatal error, in order to avoid
+// reboot loops.
+func (dn *Daemon) getPendingConfig() (string, error) {
+	journalOutput, err := exec.Command("journalctl", "-b", "-1", "-o", "json", fmt.Sprintf("MESSAGE_ID=%s", pendingStateMessageID)).CombinedOutput()
+	if err != nil {
+		return "", err
+	}
+	if len(journalOutput) == 0 {
+		return "", nil
+	}
+	lines := strings.Split(string(journalOutput), "\n")
+	if len(journalOutput) > 0 {
+		last := lines[len(lines)-1]
+		type journalMsg struct {
+			Message string `json:"MESSAGE,omitempty"`
+			BootID  string `json:"BOOT_ID,omitempty"`
+			Pending int    `json:"PENDING,omitempty"`
+		}
+		entry := &journalMsg{}
+		if err := json.Unmarshal([]byte(last), entry); err != nil {
+			return "", errors.Wrapf(err, "got %v %v %v", last, lines, journalOutput)
+		}
+		if entry.BootID == dn.bootID {
+			return "", fmt.Errorf("pending config %s bootID %s matches current! Failed to reboot?", entry.Message, dn.bootID)
+		}
+		if entry.Pending == 1 {
+			return entry.Message, nil
+		}
+		return "", nil
+	}
+	return "", fmt.Errorf("no pending config found in journal: %v", string(journalOutput))
+}
+
+func (dn *Daemon) logPendingConfig(pending *mcfgv1.MachineConfig, isPending int) error {
+	logger := exec.Command("logger", "--journald")
```

Comment:
It may be worth noting why writing to `stdin` is used rather than directly executing the command with the message. At first I was going to recommend simplifying via a direct call until I gave `logger` a run and realized that direct calling it is awkward (double enter to end input).